### PR TITLE
Unrestricted ZMP Fix

### PR DESCRIPTION
**Pull request by VHchavez**
## Description
Zmp now distinguishes between different $v_c$ in unrestricted calculations. 

## Status
- [x] Ready to go

### CODE REVIEW

File: n2v/methods/zmp.py [hunk 1]
```diff
@@ -314,7 +319,14 @@ def generate_s_functional(self, lam, zmp_functional, Cocca, Coccb, Da, Db):
             J, _ = self.form_jk(Cocca, Coccb)
 
             #Equation 7 of Reference (1)
-            vc = lam * ( (J[0] + J[1]) - (self.J0[0] + self.J0[1]) )
+            if self.ref == 1:
+                vc_a = 2 * lam * ( J[0] - self.J0[0] ) 
+                vc = [vc_a]
+            if self.ref == 2:
+                vc_a = lam * ( J[0] - self.J0[0] ) 
+                vc = [vc_a]
```
Comment (Yuming-Shi) [inline n2v/methods/zmp.py:327]: Can we delete this line

Comment (VHchavez) [reply n2v/methods/zmp.py:327]: as in just 'else'? or...

Comment (ymshi449) [reply n2v/methods/zmp.py:327]: I think we don't need vc = [vc_a]

File: n2v/methods/zmp.py [hunk 2]
```diff
@@ -314,7 +319,14 @@ def generate_s_functional(self, lam, zmp_functional, Cocca, Coccb, Da, Db):
             J, _ = self.form_jk(Cocca, Coccb)
 
             #Equation 7 of Reference (1)
-            vc = lam * ( (J[0] + J[1]) - (self.J0[0] + self.J0[1]) )
+            if self.ref == 1:
+                vc_a = 2 * lam * ( J[0] - self.J0[0] ) 
+                vc = [vc_a]
+            else:
+                vc_a = lam * ( J[0] - self.J0[0] ) 
+                vc = [vc_a]
```
Comment (ymshi449) [inline n2v/methods/zmp.py:327]: I think we dont need vc = [vc_a]

**Pull request merged**